### PR TITLE
HIVE-27962: Iceberg: Runtime predicate pushdown doesn't work

### PR DESCRIPTION
**Pull request by deniskuzZ**
### What changes were proposed in this pull request?

BugFix

### Why are the changes needed?

Runtime predicate pushdown doesn't work after HIVE-27273

### Does this PR introduce _any_ user-facing change?

No

### Is the change a dependency upgrade?

No

### How was this patch tested?

On a cluster with 1Tb dataset

### CODE REVIEW

**Review by aturoczy — COMMENTED** [review]
So, if the expression has a **BETWEEN** statement and leaf is less than 2 then returns true?
But **BETWEEN** should always contains at least 2 literals. Or maybe I do not understand correctly. 

Do you have a simple query for this?

**Review by aturoczy — APPROVED** [review]
+1

File: iceberg/iceberg-handler/src/test/java/org/apache/iceberg/mr/hive/TestHiveIcebergFilterFactory.java [hunk 1]
```diff
@@ -138,21 +138,6 @@ public void testBetweenOperand() {
     assertEquals(actual.right().op(), expected.right().op());
   }
 
-  @Test
-  public void testUnsupportedBetweenOperandEmptyLeaves() {
-    SearchArgument.Builder builder = SearchArgumentFactory.newBuilder();
-    final SearchArgument arg =
-        new MockSearchArgument(
```
Comment (zhangbutao) [inline iceberg/iceberg-handler/src/test/java/org/apache/iceberg/mr/hive/TestHiveIcebergFilterFactory.java:145]: Maybe this mock class also need to be removed.
https://github.com/apache/hive/blob/36ce858163a19e29eafe4a8d3307191bc28fc175/iceberg/iceberg-handler/src/test/java/org/apache/iceberg/mr/hive/TestHiveIcebergFilterFactory.java#L282

Comment (deniskuzZ) [reply iceberg/iceberg-handler/src/test/java/org/apache/iceberg/mr/hive/TestHiveIcebergFilterFactory.java:145]: removed

**Review by zhangbutao — APPROVED** [review]
LGTM

**Pull request merged**